File: MsgServerClientSdk/SdkMsgClientAndroid/sdkmsgclient/src/main/java/org/dync/teameeting/sdkmsgclient/msgs/MSSqlite3Manager.java
package org.dync.teameeting.sdkmsgclient.msgs;

import android.content.Context;

import java.util.ArrayList;

/**
 * Created by hp on 7/11/16.
 */
public class MSSqlite3Manager {
    private static MSSqlite mSqlite3Db = null;

    public void InitManager(Context context) {
        mSqlite3Db = new MSSqlite(context);
    }

    public void UninManager() {

    }

    public boolean IsUserExists(String userId, String seqnId) {
        return mSqlite3Db.IsUserExistsInDb(userId, seqnId);
    }

    public void AddUser(String userId) {
        mSqlite3Db.InsertSeqn(userId, userId, 0);
    }

    public void UpdateUserSeqn(String userId, long seqn) {
        mSqlite3Db.UpdateSeqn(userId, userId, seqn);
    }

    public long GetUserSeqn(String userId) {
        return mSqlite3Db.SelectSeqn(userId, userId);
    }


    public void AddGroup(String userId, String groupId) {
        mSqlite3Db.InsertGroupId(userId, groupId);
    }

    public boolean IsGroupExistsInDb(String userId, String groupId) { return mSqlite3Db.IsGroupExistsInDb(userId, groupId); }

    public void AddGroupSeqn(String userId, String grpId, long seqn) {
        mSqlite3Db.InsertSeqn(userId, grpId, seqn);
    }

    public void UpdateGroupSeqn(String userId, String grpId, long seqn) {
        mSqlite3Db.UpdateSeqn(userId, grpId, seqn);
    }

    public void UpdateGroupInfo(String userId, String grpId, long seqn, int isFetched) {
        mSqlite3Db.UpdateSeqn(userId, grpId, seqn, isFetched);
    }

    public long GetGroupSeqn(String userId, String grpId) {
        return mSqlite3Db.SelectSeqn(userId, grpId);
    }

    public ArrayList<GroupInfo> GetGroupInfo(String userId) {
        return mSqlite3Db.SelectGroupInfo(userId);
    }

    public void DelGroup(String userId, String groupId) {
        mSqlite3Db.DeleteGroupId(userId, groupId);
    }
}

File: MsgServer/MsgServerClient/ClientCommon/MSSqlite3Manager.cpp
//
//  MSSqlite3Manager.mm
//  SdkMsgClientIos
//
//  Created by hp on 7/1/16.
//  Copyright © 2016 Dync. All rights reserved.
//

#include "MSSqlite3Manager.h"

void MSSqlite3Manager::InitManager(const std::string& userId)
{
    _sqlite3Db = new MSSqlite3DB();
    if (_sqlite3Db)
    {
        _sqlite3Db->OpenDb(userId);
    }
}

void MSSqlite3Manager::UninManager()
{
    if (_sqlite3Db)
    {
        _sqlite3Db->CloseDb();
        delete _sqlite3Db;
        _sqlite3Db = nullptr;
    }
}

bool MSSqlite3Manager::IsUserExists(const std::string& userId, const std::string& seqnId)
{
    return _sqlite3Db->IsUserExistsInDb(userId, seqnId);
}

bool MSSqlite3Manager::IsGroupExists(const std::string& userId, const std::string& grpId)
{
    return _sqlite3Db->IsGroupExists(userId, grpId);
}

void MSSqlite3Manager::AddUser(const std::string& userId)
{
    if(_sqlite3Db)
    {
        _sqlite3Db->InsertSeqn(userId, userId, 0);
    }
}

void MSSqlite3Manager::UpdateUserSeqn(const std::string& userId, ::google::protobuf::int64 seqn)
{
    if (_sqlite3Db)
    {
        _sqlite3Db->UpdateSeqn(userId, userId, seqn);
    }
}

void MSSqlite3Manager::GetUserSeqn(const std::string& userId, ::google::protobuf::int64* seqn)
{
    if (_sqlite3Db)
    {
        _sqlite3Db->SelectSeqn(userId, userId, seqn);
    }
}

void MSSqlite3Manager::AddGroupId(const std::string& userId, const std::string& groupId)
{
    if (_sqlite3Db)
    {
        _sqlite3Db->InsertGroupId(userId, groupId);
    }
}

void MSSqlite3Manager::AddGroupSeqn(const std::string& userId, const std::string& grpId, ::google::protobuf::int64 seqn)
{
    if (_sqlite3Db)
    {
        _sqlite3Db->InsertSeqn(userId, grpId, seqn);
    }
}

void MSSqlite3Manager::UpdateGroupSeqn(const std::string& userId, const std::string& grpId, ::google::protobuf::int64 seqn)
{
    if (_sqlite3Db)
    {
        _sqlite3Db->UpdateSeqn(userId, grpId, seqn);
    }
}

void MSSqlite3Manager::UpdateGroupInfo(const std::string& userId, const std::string& grpId, ::google::protobuf::int64 seqn, int isfetched)
{
    if (_sqlite3Db)
    {
        _sqlite3Db->UpdateSeqnWithFetched(userId, grpId, seqn, isfetched);
    }
}

void MSSqlite3Manager::GetGroupSeqn(const std::string& userId, const std::string& grpId, ::google::protobuf::int64* seqn)
{
    if (_sqlite3Db)
    {
        _sqlite3Db->SelectSeqn(userId, grpId, seqn);
    }
}

void MSSqlite3Manager::GetGroupInfo(const std::string& userId, DbResultVector& info)
{
    if (_sqlite3Db)
    {
        _sqlite3Db->SelectGroupInfo(userId, info);
    }
}

void MSSqlite3Manager::DelGroupId(const std::string& userId, const std::string& groupId)
{
    if (_sqlite3Db)
    {
        _sqlite3Db->DeleteSeqn(userId, groupId);
        _sqlite3Db->DeleteGroupId(userId, groupId);
    }
}
